src/SourceLocator/Located/LocatedSource.php

```diff
@@ -32,8 +31,6 @@ public function __construct(private string $source, private string|null $name, s
         if ($filename !== null) {
             assert($filename !== '');
 
-            FileChecker::assertReadableFile($filename);
```

Comment:
Worth using realpath here?

Reply:
I did not find a single place where we pass in filename, but did not already do IO on it, like e.g. `file_get_contents`.
I think this check here is useless.

we could hide it behand a `if(zend-assertions-enabled)` switch or similar, if you prefer.

I don't get why `realpath()` would be different then what we have right now.

Reply:
This API is public, so yah, some invariant checking would be expected.

Reply:
Ah, never mind, it's `@internal`

I'll 🚢 as-is, thanks @staabm!

Reply:
Thank you 🙏